k8s/eks/bash/functions.sh

```diff
@@ -0,0 +1,346 @@
+#!/bin/bash
+
+# error log prep check
+prep_log_dir(){ 
+  mkdir -p $real_path/log/$start-log/
+  mkdir -p $real_path/.cluster/
+}
+
+concat_availability_zone(){
+  AVAILABILITY_ZONE=$REGION$AZ_SUFFIX
+}
+
+create_cluster() {
+  if test -f "$real_path/.cluster/$CLUSTER_NAME.cs"; then
+    echo "File $real_path/.cluster/$CLUSTER_NAME.cs exists."
+    echo "You may want to run uninstall script first!"
+    exit
+fi
+  eksctl create cluster --name $CLUSTER_NAME --without-nodegroup --region=$REGION
+  echo "cluster_created=true" >> $real_path/.cluster/$CLUSTER_NAME.cs
+  echo "cluster_name=$CLUSTER_NAME" >> $real_path/.cluster/$CLUSTER_NAME.cs
+  echo "region=$REGION" >> $real_path/.cluster/$CLUSTER_NAME.cs
+}
+
+deploy_multus_ds() {
+  kubectl apply -f $real_path/yaml/multus-daemonset.yml
+}
+
+deploy_weave_cni() {
+  #kubectl apply -f "https://cloud.weave.works/k8s/net?k8s-version=$(kubectl version | base64 | tr -d '\n')"
+  kubectl apply -f https://github.com/weaveworks/weave/releases/download/v2.8.1/weave-daemonset-k8s-1.11.yaml
+}
+
+create_weave_networkattachmentdefinition() {
+  kubectl create -f $real_path/yaml/weave.yml
+}
+
+deploy_cluster_role_binding() {
+  kubectl create clusterrolebinding service-reader --clusterrole=service-reader --serviceaccount=default:default
+  kubectl create -f $real_path/yaml/clusterrolebinding.yml
+}
+
+weave_ip_tables_drop() {
+  kubectl create -f $real_path/yaml/drop-weave-cm.yml
+  kubectl create -f $real_path/yaml/drop-weave-ds.yml
+}
+
+multus_softlink() {
+  kubectl create -f $real_path/yaml/softlink-cm.yml
+  kubectl create -f $real_path/yaml/softlink-ds.yml
+}
+
+obtain_ami_id(){
+  AMI_ID=$(aws ec2 describe-images --filters "Name=name,Values=$AMI_NAME" --owners amazon --region $REGION | jq -r '.Images[0].ImageId')
+}
+
+make_cluster_config(){
+obtain_ami_id
+concat_availability_zone
+    cat <<EOT >> $real_path/$CLUSTER_NAME.yaml
```

Comment:
As said before, `$CLUSTER_NAME.yaml` has to be gitignored.

Reply:
Added to gitignore.